modules/popup/react/lib/usePopupStack.ts

```diff
@@ -4,27 +4,42 @@ import {PopupStack} from '@workday/canvas-kit-popup-stack';
 
 /**
  * This hook will add a `ref` element to the Popup stack on mount and remove on unmount. If you use
- * `Popper`, the popper `ref` is automatically added/removed from the Popup stack. The Popup stack is
- * required for proper z-index values to ensure Popups are rendered correct. It is also required for
- * global listeners like click outside or escape key closing a popup. Without the Popup stack, all
- * popups will close rather than only the topmost one.
- * @param ref Ref of the element that is considered the container for a Popup. Usually the `Popup` component
- * @param target Usually the trigger of a popup. This will fix `bringToTop` and should be provided by all
- * ephemeral-type popups (like Tooltips, Select menus, etc). It will also add in clickOutside detection
+ * `Popper`, the popper `ref` is automatically added/removed from the Popup stack. The Popup stack
+ * is required for proper z-index values to ensure Popups are rendered correct. It is also required
+ * for global listeners like click outside or escape key closing a popup. Without the Popup stack,
+ * all popups will close rather than only the topmost one.
+ * @param ref This ref will be managed by the PopupStack and should not be managed by React. Do not
+ * apply this ref to a React element. Doing so will result in an error. Instead, use this `ref`
+ * directly with `ReactDOM.createPortal(<YourComponent>, ref.current!)`. This is definitely strange
+ * for React code, but is necessary for the PopupStack to remain framework agnostic and flexible to
+ * integrate with existing popup stack systems
+ * @param target Usually the trigger of a popup. This will fix `bringToTop` and should be provided
+ * by all ephemeral-type popups (like Tooltips, Select menus, etc). It will also add in clickOutside
+ * detection
  */
 export const usePopupStack = <E extends HTMLElement>(
   ref: React.RefObject<E>,
```

Comment:
Should this even be a `RefObject` anymore? The Popup Stack controls this element now. I'm not even sure we should use a ref... Changing this would be an obvious breaking change. Not changing it is a subtle breaking change.